### PR TITLE
Attempt to parse `SWIFT_VERSION` from xcconfig during target inspection

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/7731

- [x] specs

### CODE REVIEW

**Comment by dnkoutso** [comment]
Meh it appears this assumes the file exists...

File: lib/cocoapods/installer/analyzer/target_inspector.rb [hunk 1]
```diff
@@ -218,7 +218,10 @@ def compute_recommends_frameworks(target_definition, native_targets)
         #
         def compute_swift_version_from_targets(targets)
           versions_to_targets = targets.inject({}) do |memo, target|
-            versions = target.resolved_build_setting('SWIFT_VERSION').values
+            resolve_against_xcconfig = target.build_configuration_list.build_configurations.all? do |bc|
```
Comment (dnkoutso) [inline lib/cocoapods/installer/analyzer/target_inspector.rb]: yes there is a case where the xcconfig is not set for all the configuration lists of the target but meh

Comment (segiddins) [reply lib/cocoapods/installer/analyzer/target_inspector.rb]: this is kinda gross, please add a comment ?

**Comment by dnkoutso** [comment]
GH bug seems to prevent me from responding to your comment. Will add a comment and merge on green.

**Pull request merged**